qcodes/instrument_drivers/tektronix/Keithley_3706A_Matrix_channels.py

```diff
@@ -0,0 +1,709 @@
+        return channels_to_connect
+
+    def get_idn(self) -> Dict[str, Optional[str]]:
+        """
+        Overwrites the generic QCoDeS get IDN method. Returns
+        a dictionary including the vendor, model, serial number and
+        firmware version of the instrument.
+        """
+        idnstr = self.ask_raw('*IDN?')
+        vendor, model, serial, firmware = map(str.strip, idnstr.split(','))
+        model = model[6:]
+
+        idn: Dict[str, Optional[str]] = {'vendor': vendor, 'model': model,
+                                         'serial': serial, 'firmware': firmware}
+        return idn
+
+    def get_switch_cards(self) -> Tuple[Dict[str, str], ...]:
+        """
+        Returns a list of dictionaries listing the properties of the installed
+        switch cards including the slot number tha it is installed, model,
+        firmware version and serial number.
+        """
+        switch_cards: List[Dict[str, str]] = []
+        for i in range(1, 7):
+            scard = self.ask(f'slot[{i}].idn')
+            if scard != 'Empty Slot':
+                model, mtype, firmware, serial = map(str.strip,
+                                                     scard.split(','))
+                sdict = {'slot_no': str(i), 'model': model, 'mtype': mtype,
+                         'firmware': firmware, 'serial': serial}
+                switch_cards.append(sdict)
+        return tuple(switch_cards)
+
+    def get_available_memory(self) -> Dict[str, Optional[str]]:
+        """
+        Returns the amount of memory that is currently available for
+        storing scripts, configurations and channel patterns.
+        """
+        memstring = self.ask('memory.available()')
+        system_memory, script_memory, \
+            pattern_memory, config_memory = map(str.strip, memstring.split(','))
+
+        memory_available: Dict[str, Optional[str]] = {
+            'System Memory  (%)': system_memory,
+            'Script Memory  (%)': script_memory,
+            'Pattern Memory (%)': pattern_memory,
+            'Config Memory  (%)': config_memory
+        }
+        return memory_available
+
+    def get_interlock_state(self) -> Tuple[Dict[str, str], ...]:
+        """
+        A function that collects the interlock status of the installed cards.
+        The channel relays can continue to operate even if the interlock
+        in the corresponding slot is disengaged, one cannot perform
+        measurements through the switching card, as the analog backplanes
+        cannot be energized.
+        """
+        slot_id = self._get_slot_id()
+        interlock_status = {0: 'Interlock is disengaged',
+                            1: 'Interlock is engaged'}
+        states: List[Dict[str, str]] = []
+        for i in slot_id:
+            state = self.ask(f'slot[{int(i)}].interlock.state')
+            state_dict = {'slot_no': i,
+                          'state': interlock_status[int(float(state))]}
```

Comment:
is the `int(state)` not enough because the instrument returns `1.0` instead of `1`? :)

Reply:
Instrument returns a string "1.0".

Reply:
ok, sad but true